### ISSUE TITLE
Set initial sort params

### ISSUE THREAD

**Issue by joegaudet** [miguelcobain/ember-yeti-table#284]
Would be handy when providing pre-sorted data, or data that comes from an async source to define the initial sorting in some way. Especially for routing query parameters to the yeti table.

**Comment by cah-briangantzler** [comment]
```
<header.column @prop="points" @sort="asc" as |column|>
  Points
</header.column>
```

This is hardcoded as the initial sort, but you could use a variable if you like.

https://miguelcobain.github.io/ember-yeti-table/docs/sorting

**Comment by arthur5005** [comment]
Yeah, we saw that, but having to do this work on every column definition is a bit arduous. 

We ended up extending the table and and overwriting the registerColumn method to set the column's initial state. This example assumes you can only sort 1 column at a time and it follows the `prop`\`-prop` way of defining ascending and descending. 

```
  registerColumn(column) {
    if (this.sort && this.sort.includes(column.prop)) {
      column.set('sort', this.sort.includes('-') ? 'desc' : 'asc');
    }

    super.registerColumn(column);
  }
```

A sorting event would also be helpful for setting the QP

Not sure what a more general version of this might look like, but happy to contribute, really like the ergonomics of this table component for the most part.

**Comment by miguelcobain** [comment]
> Yeah, we saw that, but having to do this work on every column definition is a bit arduous.

I don't quite follow. You just have to do this for one column. The one you want sorted.
How is that different from specifying a `@sort` argument on the table?

Wait, do you need to sort based on a query parameter?

**Comment by arthur5005** [comment]
Yes, we are driving the state of the table via Ember query parameters for both sort and pagination. 
This makes it easy for our users to share links and see exactly the same thing (after sort and pagination).

We could, as you said, on a per column definition basis, set the sort state based on the query parameter variable, using a template helper, but it feels like a lot of plumbing, I think that's all we're trying to say.

**Comment by miguelcobain** [comment]
I see. I think an optional global `@sort` argument + `@onSortChanged` action would make sense.

**Comment by arthur5005** [comment]
Yeah, us too. We're trying to understand what the API of the @sort property looks like more generically. Does YetiTable support or is intending to support `multiple` sorts?

**Comment by miguelcobain** [comment]
YetiTable does support multiple sorts, etc. So this api would have to account for that.

The first idea that comes to mind is to use a comma separated string with all the props in some format. Either prepending `-` or appending `:asc`/`:desc`.

Simple single sort:

```hbs
@sort="firstName:asc"
```

if no direction is provided, `asc` is the default, so the above is functionally equivalent to:
```hbs
@sort="firstName"
```

This would sort by firstName ascending first and then by lastName descending:
```hbs
@sort="firstName:asc,lastName:desc"
```

after splitting the string by `,`, we need to trim it, so that this also becomes a valid `@sort` value:
```hbs
@sort="    firstName:asc   ,   lastName:desc  "
```

It makes sense to also support the array itself, especially with the introduction of the ember's `array` helper:

```hbs
@sort={{array "firstName" "lastName:desc"}}
```

Using the `-` prefix, the above examples would become:

```hbs
@sort="firstName"
```

```hbs
@sort="firstName,-lastName"
```

```hbs
@sort="    firstName   ,   -lastName  "
```

```hbs
@sort={{array "firstName" "-lastName"}}
```

I really like the ergonomics of the `-` prefix. However the first format seems more in line with the existing api on the columns. 🤔 

```hbs
<header.column @prop="points" @sort="asc">
  Points
</header.column>
```

Any thoughts?

**Comment by joegaudet** [comment]
My vote would be for the - prefix as that's consistent with JSON:API (the
whole value of this feature is piping query params to an API call).

.joe

**Comment by cah-briangantzler** [comment]
So we actually ran a direct need for this and so better understand the issue. We created some custom code to handle this. Once I am done converting the components to glimmer, will make a PR that would make this easier.
